specs/agents/tracing-spans-service-target.md

```diff
@@ -0,0 +1,236 @@
+
+### Field values
+
+- `span.context.service.target.*` fields should be omitted for non-exit spans.
+- Values set by user through the agent API should have priority over inferred values.
+- `span.context.service.target.type` should have the same value as `span.subtype` and fallback to `span.type`.
+- `span.context.service.target.name` depends on the span context attributes
+
+On agents, the following algorithm should be used to infer the values for `span.context.service.target.*` fields. 
+```javascript
+// span created on agent
+span = {};
+
+if (span.isExit) {
+  service_target = span.context.service.target;
+  
+  if (!service_target.type) { // infer type from span type & subtype
+      
+    // use sub-type if provided, fallback on type othewise
+    service_target.type = span.subtype || span.type;
+  }
+  
+  if (!service_target.name) { // infer name from span attributes
+      
+    if (span.context.db.instance) {  // database spans
+      service_target.name = span.context.db.instance;
+
+    } else if (span.context.message) { // messaging spans
+      service_target.name = span.context.message.queue.name
+
+    } else if (context.http.url) { // http spans
+      service_target.name = context.http.host;
+      if (context.http.url.port > 0) {
+        service_target.name += ":" + context.http.port;
+      }
+    }
+
+  }
+} else {
+    // non-exit spans should not have service.target.* fields
+    span.context.service.target = undefined;
+}
+```
+
+The values for `span.context.db.instance` are described in [SQL Databases](./tracing-instrumentation-db.md#sql-databases).
+
+The values for `span.context.message.queue.name` are described in [Messaging context fields](./tracing-instrumentation-messaging.md#context-fields)
+
+### User API
+
+Agents SHOULD provide an API entrypoint to set the value of `span.context.destination.service.resource`,
+setting an empty or `null` value allows the user to discard the inferred value.
+This API entrypoint should be marked as deprecated and replaced by the following:
+
+Agents SHOULD provide an API entrypoint to set the value of `span.context.service.target.type` and `span.context.service.target.name`,
+setting an empty or `null` value on both of those fields allows the user to discard the inferred values.
+
+When a user-provided value is set, it should take precedence over inferred values from the span `_.type` `_.subtype` or any `_.context` attribute.
+
+### Phase 3
+
+In Phase 3 the `span.destination.service.resource` field value is inferred on APM Server with the following algorithm:
+
+```javascript
+
+// span from agent intake
+span = {};
+
+// inferred destination resource (if any)
+destination_resource = undefined;
+
+if (span.service.target.type) {
+  // new fields, store them as-provided
+  // infer resource destination from new fields
+  destination_resource = span.service.target.type;
+  if (span.service.target.name) {
```

Comment:
By this logic, if only `span.service.target.name` is defined then the `span.destination.service.resource` might not get populated or `span.service.target.type` might be overridden by some other value depending on span.type.

Are these effects expected or is there a strict requirement of providing `target.type` if `target.name` is provided?

Reply:
Got the answer, we will ignore `target.name` if `target.type` is not specified. Resolving this.